### PR TITLE
Support for converting to Kubernetes

### LINKED ISSUE #7: Add support for Kubernetes

### PR DESCRIPTION
**Pull request by kadel**
fixes #7

### CODE REVIEW

**Comment by kadel** [comment]
It is quite ugly, but I didn't wan to spend much time polishing code, if we are not sure about structure.

**Comment by kadel** [comment]
Merge this after #1. 
It will have conflicts.

**Comment by kadel** [comment]
@rtnpro @surajssd can please you look at this?

File: pkg/transformers/main.go [hunk 1]
```diff
@@ -3,21 +3,29 @@ package transformers
...
+	case "openshift":
+		list, err := openshift.Transform(paths...)
 		if err != nil {
 			return err
 		}
-		openshift.Print(obj)
-	} else {
-		err := fmt.Errorf("Provider not supported")
-		return err
+		kubernetes.PrintList(list)
```
Comment (rtnpro) [inline pkg/transformers/main.go]: :+1: for using same code for printing.

File: pkg/transformers/openshift/openshift.go [hunk 2]
```diff
@@ -19,50 +14,21 @@ import (
...
 		return nil, err
 	}
 
-	// Convert template objects to versioned objects
-	var convErr error
-	template.Objects, convErr = convertToVersion(template.Objects, "v1")
-	if convErr != nil {
-		panic(convErr)
-	}
+	list := ConvertToList(template)
```
Comment (rtnpro) [inline pkg/transformers/openshift/openshift.go]: :+1: for normalizing the return types to a common type for both ``openshift`` and ``kubernetes``

File: pkg/transformers/kubernetes/generate.go [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+}
+
+func rangeToPort(s string) string {
+	parts := strings.SplitN(s, "-", 2)
+	return parts[0]
+}
+
+// warnUnusableComposeElements add warnings for unsupported elements in the provided service config
+func warnUnusableComposeElements(k string, v *project.ServiceConfig, warnings map[string][]string) {
+	fn := func(msg string) {
```
Comment (rtnpro) [inline pkg/transformers/kubernetes/generate.go]: Should we not show warning for ``build`` for Kubernetes? I do not see a way how to support builds for k8s at the moment. There's a long ongoing issue for it: https://github.com/kubernetes/kubernetes/issues/1293

**Comment by rtnpro** [comment]
:+1: LGTM

File: pkg/transformers/kubernetes/generate.go (hunk 3 above)
Comment (kadel) [inline pkg/transformers/kubernetes/generate.go]: Right now if there is build it erros out during conversion to kubernetes. see: https://github.com/kadel/henge/blob/add-kubernetes/pkg/transformers/kubernetes/generate.go#L63,L64

My thought was that in future Henge might be able to build image and push it to user provided registry.

**Comment by pradeepto** [comment]
Awesome! :)

**Pull request merged**